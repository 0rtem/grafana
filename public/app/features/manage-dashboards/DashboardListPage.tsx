import React, { PureComponent } from 'react';
import { hot } from 'react-hot-loader';
import { connect } from 'react-redux';
import { DashboardSection, NavModel } from 'app/types';
import PageHeader from 'app/core/components/PageHeader/PageHeader';
import ActionBar from './ActionBar';
import ActiveFilters from './ActiveFilters';
import Section from './Section';
import SectionActions from './SectionActions';
import { getNavModel } from 'app/core/selectors/navModel';
import { loadSections } from './state/actions';
import { getSections, getFolderId, getHasFilters } from './state/selectors';

export interface Props {
  navModel: NavModel;
  hasFilters: boolean;
  sections: DashboardSection[];
  folderId: number;
  loadDashboardListItems: typeof loadSections;
}

export class DashboardListPage extends PureComponent<Props, any> {
  componentDidMount() {
    this.fetchSections();
  }

  async fetchSections() {
    await this.props.loadDashboardListItems();
  }

  render() {
    const { navModel, folderId, hasFilters, sections } = this.props;

    return (
      <div>
        <PageHeader model={navModel} />
        <div className="page-container page-body">
          <div className="dashboard-list">
            <ActionBar />
            {hasFilters && <ActiveFilters />}
            {hasFilters &&
              sections.length !== 0 && (
                <div className="search-results">
                  <em className="muted">No dashboards matching your query were found.</em>
                </div>
              )}
            {folderId !== null &&
              !hasFilters &&
              sections.length === 0 && (
                <div className="search-results">
                  <em className="muted">No dashboards found.</em>
                </div>
              )}
            {sections.length > 0 && <SectionActions />}
            <div className="search-results-container">
              {sections.length > 0 &&
                sections.map((section, index) => {
                  return <Section section={section} key={index} />;
                })}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  const manageDashboardsState = state.manageDashboards;

  return {
    navModel: getNavModel(state.navIndex, 'manage-dashboards'),
    hasFilters: getHasFilters(manageDashboardsState),
    sections: getSections(state.sections),
    folderId: getFolderId(manageDashboardsState),
  };
}

const mapDispatchToProps = {
  loadDashboardListItems: loadSections,
};

export default hot(module)(connect(mapStateToProps, mapDispatchToProps)(DashboardListPage));
