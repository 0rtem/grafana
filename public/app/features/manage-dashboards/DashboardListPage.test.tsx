import React from 'react';
import { shallow } from 'enzyme';
import { DashboardListPage, Props } from './DashboardListPage';

const setup = (propOverrides?: object) => {
  const props: Props = {};

  const wrapper = shallow(<DashboardListPage {...props} />);
  const instance = wrapper.instance() as DashboardListPage;

  return {
    wrapper,
    instance,
  };
};
