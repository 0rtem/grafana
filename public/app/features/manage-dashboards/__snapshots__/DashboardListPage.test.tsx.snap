// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div>
  <PageHeader
    model={Object {}}
  />
  <div
    className="page-container page-body"
  >
    <div
      className="dashboard-list"
    >
      <Connect(ActionBar) />
      <div
        className="search-results"
      >
        <em
          className="muted"
        >
          No dashboards found.
        </em>
      </div>
      <div
        className="search-results-container"
      />
    </div>
  </div>
</div>
`;

exports[`Render should render sections 1`] = `
<div>
  <PageHeader
    model={Object {}}
  />
  <div
    className="page-container page-body"
  >
    <div
      className="dashboard-list"
    >
      <Connect(ActionBar) />
      <Connect(SectionActions) />
      <div
        className="search-results-container"
      >
        <Connect(Section)
          key="Such a cool folder-0"
          section={
            Object {
              "checked": false,
              "expanded": false,
              "hideHeader": false,
              "icon": "cool-icon",
              "id": 1,
              "items": Array [],
              "score": 1,
              "title": "Such a cool folder",
              "uid": "asdf1",
              "url": "some/url/1",
            }
          }
        />
        <Connect(Section)
          key="Such a cool folder-1"
          section={
            Object {
              "checked": false,
              "expanded": false,
              "hideHeader": false,
              "icon": "cool-icon",
              "id": 2,
              "items": Array [],
              "score": 2,
              "title": "Such a cool folder",
              "uid": "asdf2",
              "url": "some/url/2",
            }
          }
        />
        <Connect(Section)
          key="Such a cool folder-2"
          section={
            Object {
              "checked": false,
              "expanded": false,
              "hideHeader": false,
              "icon": "cool-icon",
              "id": 3,
              "items": Array [],
              "score": 3,
              "title": "Such a cool folder",
              "uid": "asdf3",
              "url": "some/url/3",
            }
          }
        />
        <Connect(Section)
          key="Such a cool folder-3"
          section={
            Object {
              "checked": false,
              "expanded": false,
              "hideHeader": false,
              "icon": "cool-icon",
              "id": 4,
              "items": Array [],
              "score": 4,
              "title": "Such a cool folder",
              "uid": "asdf4",
              "url": "some/url/4",
            }
          }
        />
        <Connect(Section)
          key="Such a cool folder-4"
          section={
            Object {
              "checked": false,
              "expanded": false,
              "hideHeader": false,
              "icon": "cool-icon",
              "id": 5,
              "items": Array [],
              "score": 5,
              "title": "Such a cool folder",
              "uid": "asdf5",
              "url": "some/url/5",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Render should show filters if true 1`] = `
<div>
  <PageHeader
    model={Object {}}
  />
  <div
    className="page-container page-body"
  >
    <div
      className="dashboard-list"
    >
      <Connect(ActionBar) />
      <Connect(ActiveFilters) />
      <div
        className="search-results"
      >
        <em
          className="muted"
        >
          No dashboards matching your query were found.
        </em>
      </div>
      <div
        className="search-results-container"
      />
    </div>
  </div>
</div>
`;

exports[`Render should show no matching sections to filtering 1`] = `
<div>
  <PageHeader
    model={Object {}}
  />
  <div
    className="page-container page-body"
  >
    <div
      className="dashboard-list"
    >
      <Connect(ActionBar) />
      <Connect(ActiveFilters) />
      <Connect(SectionActions) />
      <div
        className="search-results-container"
      >
        <Connect(Section)
          key="Such a cool folder-0"
          section={
            Object {
              "checked": false,
              "expanded": false,
              "hideHeader": false,
              "icon": "cool-icon",
              "id": 1,
              "items": Array [],
              "score": 1,
              "title": "Such a cool folder",
              "uid": "asdf1",
              "url": "some/url/1",
            }
          }
        />
        <Connect(Section)
          key="Such a cool folder-1"
          section={
            Object {
              "checked": false,
              "expanded": false,
              "hideHeader": false,
              "icon": "cool-icon",
              "id": 2,
              "items": Array [],
              "score": 2,
              "title": "Such a cool folder",
              "uid": "asdf2",
              "url": "some/url/2",
            }
          }
        />
        <Connect(Section)
          key="Such a cool folder-2"
          section={
            Object {
              "checked": false,
              "expanded": false,
              "hideHeader": false,
              "icon": "cool-icon",
              "id": 3,
              "items": Array [],
              "score": 3,
              "title": "Such a cool folder",
              "uid": "asdf3",
              "url": "some/url/3",
            }
          }
        />
        <Connect(Section)
          key="Such a cool folder-3"
          section={
            Object {
              "checked": false,
              "expanded": false,
              "hideHeader": false,
              "icon": "cool-icon",
              "id": 4,
              "items": Array [],
              "score": 4,
              "title": "Such a cool folder",
              "uid": "asdf4",
              "url": "some/url/4",
            }
          }
        />
        <Connect(Section)
          key="Such a cool folder-4"
          section={
            Object {
              "checked": false,
              "expanded": false,
              "hideHeader": false,
              "icon": "cool-icon",
              "id": 5,
              "items": Array [],
              "score": 5,
              "title": "Such a cool folder",
              "uid": "asdf5",
              "url": "some/url/5",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
